pkg/network/infraconfigurators/bridge.go

```diff
@@ -63,6 +62,11 @@ func (b *BridgePodNetworkConfigurator) DiscoverPodNetworkInterface(podIfaceName
 	}
 
 	b.tapDeviceName = generateTapDeviceName(podIfaceName)
+
+	b.vmMac, err = retrieveMacAddressFromVMISpecIface(b.vmiSpecIface)
+	if err != nil {
+		return err
+	}
```

Comment:
I am against calling this function here - this function is meant to discover the link's attributes - mac, ips, routes. 

This is the mac address as specified in the vmi spec - it is more suited to be set from the constructor (as, for instance, the tap device name we generate for some bindings).

Reply:
It is not accurate that this method "is meant to discover the link's attributes".
For example- in `masquerade.DiscoverPodNetworkInterface` - `configureIPv4Addresses` and `configureIPv6Addresses` has nothing to do with the link.

By the way, `generateTapDeviceName` is actually invoked in this specific example of the bridge discover (masquerade has another behaviour, I will add a commit to fix it).

I find it much more convenient to have one place that calculates the mac address over starting to search where the existing value on `b.vmMac` came from.

Reply:
Regarding - "I will add a commit to fix it". Now I realize I cannot since in bridge binding the original interface name is changed. Since we want the tap device name to be generated from the original pod interface, it should be done before the swap with the dummy.

Reply:
Yes. Which, imo, makes it a candidate to be created in the constructor, but I won't insist.

I guess the important discussion here is what should a method named `DiscoverPodNetworkInterface` do. 
I would expect it to discover the pod network interface (routes / link / addresses). 

You're right in the sense that other bindings are not strictly discovering the pod network interface. IMO, they're wrong; that sort of information should be done by the constructor.

I thought this was meant to be:
  - all sorts of static information: calculated in the constructor
  - all sorts of network related info (addresses, link, routes): accessed via netlink & added to the struct on the discover call

Up to you, I won't insist in this.

Reply:
At least for me it would be much easier to read the code if the struct parameters are initialized only in one place.
I'm ok with initializing in the constructor non "netlink" related info, as long as the initialized value won't be overridden by `DiscoverPodNetworkInterface`.